dashboard/app/models/teacher_feedback.rb

```diff
@@ -150,6 +154,51 @@ def summarize(is_latest = false)
     }
   end
 
+  # Passing level, script_level and student into summarize_for_csv since it's
+  # called in a loop to avoid N+1s
+  def summarize_for_csv(level, script_level, student, sublevel_index = nil)
+    rubric_performance_json_to_ruby = {
+      performanceLevel1: "rubric_performance_level_1",
+      performanceLevel2: "rubric_performance_level_2",
+      performanceLevel3: "rubric_performance_level_3",
+      performanceLevel4: "rubric_performance_level_4"
+    }
+
+    rubric_performance_headers = {
+      performanceLevel1: "Extensive Evidence",
+      performanceLevel2: "Convincing Evidence",
+      performanceLevel3: "Limited Evidence",
+      performanceLevel4: "No Evidence"
+    }
+
+    review_state_labels = {
+      keepWorking: "Needs more work",
+      completed: "Reviewed, completed",
+    }
+
+    review_state_label = awaiting_teacher_review?(true) ? "Waiting for review" : review_state_labels[review_state&.to_sym]
+    level_num = script_level.position.to_s
+
+    if sublevel_index
+      alphabet = ('a'..'z').to_a
+      level_num += alphabet[sublevel_index]
+    end
+
+    {
+      studentName: student.name,
+      lessonNum: script_level.lesson.relative_position.to_s,
+      lessonName: script_level.lesson.localized_title,
```

Comment:
Mostly curious -- do these references to `.lesson` trigger additional queries?

Reply:
Do you mean that `script_level.lesson` is called on line 189 and 190? No. This was one of the things I was testing when I wrote the query count test https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/test/controllers/api/v1/assessments_controller_test.rb#L693, there must be some kind of built in rails magic caching because we don't perform 2 queries for that